horovod/common/logging.cc

```diff
@@ -0,0 +1,91 @@
+#include <chrono>
+#include <algorithm>
+
+#include "logging.h"
+
+namespace horovod {
+namespace common {
+
+LogMessage::LogMessage(const char* fname, int line, LogLevel severity)
+    : fname_(fname), line_(line), severity_(severity) {}
+
+void LogMessage::GenerateLogMessage(bool log_time) {
+  if (log_time) {
+    auto now = std::chrono::system_clock::now();
+    auto as_time_t = std::chrono::system_clock::to_time_t(now);
+
+    auto duration = now.time_since_epoch();
+    auto seconds = std::chrono::duration_cast<std::chrono::seconds>(duration);
+    auto micros_remainder = std::chrono::duration_cast<std::chrono::microseconds>(duration - seconds);
+
+    const size_t time_buffer_size = 30;
+    char time_buffer[time_buffer_size];
+    strftime(time_buffer, time_buffer_size, "%Y-%m-%d %H:%M:%S",
```

Comment:
Can simplify using https://stackoverflow.com/a/10289094

Reply:
I tried this and it works with gcc/g++ 5. But seeing a comment on the answer, I tried gcc/g++ 4.8. It fails to build with 4.8. In the interest of broader compatibility, I think we should not use `put_time`. What do you think?

Reply:
Makes sense. Let's stick to current implementation then.